### PR TITLE
Feature/initial architecture

### PR DESCRIPTION
**Pull request by dubhcait**
Created client folder with create-react-app, added proxy to json. Created minimal express backend 2 endpoints, one to test server is running and other to serve the client folder

### CODE REVIEW

**Comment by engshorouq** [comment]
- you push the `node_module`. you can move the `.gitignore` from the `sever` folder  to the root directory to prevent push it in the GitHub

File: src/app.js [hunk 1]
```diff
@@ -0,0 +1,30 @@
+const express = require("express");
+const path = require("path");
+const bodyParser = require("body-parser");
```
Comment (engshorouq) [inline src/app.js]: we don't need `body-parser` because it is building in express
```
app.use(express.json());
```

File: src/index.js [hunk 2]
```diff
@@ -0,0 +1,5 @@
+const app = require("./app");
+
+app.listen(app.get("PORT"), () => {
+  console.log("The magic is happening on port", app.get("PORT"));
```
Comment (engshorouq) [inline src/index.js]: put the link of server
```suggestion
  console.log(`the server is running at http://localhost:${app.get('PORT')}`);
```

File: src/app.js [hunk 3]
```diff
@@ -0,0 +1,30 @@
...
+    res.send({ express: " YOUR BACKEND IS CONNECTED" });
+  });
+
+  app.get("*", (req, res) => {
+    res.sendFile(path.join(__dirname, "..", "client", "build", "index.html"));
+  });
+
+  app.set("PORT", process.env.PORT || 9000);
+
+module.exports = app;
```
Comment (engshorouq) [inline src/app.js]: leave a line at the end of all files of `js`

File: src/app.js (hunk 1 above)
Comment (dubhcait) [reply src/app.js]: habit, thanks

**Pull request merged**